sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -59,36 +60,36 @@ class AzureKeyCredential:
     """Credential type used for authenticating to an Azure service.
     It provides the ability to update the key without creating a new client.
 
-    :param str key: The key used to authenticate to an Azure service
+    :param TKeys key: The key used to authenticate to an Azure service
```

Comment:
Is it worth making the entire class a generic?
That should improve the typing experience if we expect the type used to remain consistent throughout, e.g.
```python
class MetricsAdvisorKeyCredential(AzureKeyCredential[Tuple[str, str]]):
    ...
```

Reply:
Also - I would avoid using the `TypeVar` in the docstring, I don't think Sphinx expands this or includes the typevar information, so wouldn't tell users much. Maybe just stick with `Union[str, Tuple[str, ...]]`

Reply:
Not sure if I understand it correctly, we only want to allow str & Tuple[str, ...], if we use generic class, will this imply users that it may also accept int, boolean, etc.?

Reply:
I don't think so, because the TypeVar is still bound to only a specific set of types.